atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformerJS.kt

```diff
@@ -0,0 +1,310 @@
+            }
+
+            "getAndDecrement\$atomicfu\$long" -> {
+                code = "(function(scope) {var oldValue = $f; $f = $f.dec(); return oldValue;})"
+            }
+
+            "getAndAdd\$atomicfu" -> {
+                val arg = args[0].toSource()
+                code = "(function(scope) {var oldValue = $f; $f += $arg; return oldValue;})"
+            }
+
+            "getAndAdd\$atomicfu\$long" -> {
+                val arg = args[0].toSource()
+                code = "(function(scope) {var oldValue = $f; $f = $f.add($arg); return oldValue;})"
+            }
+
+            "addAndGet\$atomicfu" -> {
+                val arg = args[0].toSource()
+                code = "(function(scope) {$f += $arg; return $f;})"
+            }
+
+            "addAndGet\$atomicfu\$long" -> {
+                val arg = args[0].toSource()
+                code = "(function(scope) {$f = $f.add($arg); return $f;})"
+            }
+
+            "incrementAndGet\$atomicfu" -> {
+                code = "(function(scope) {return ++$f;})"
+            }
+
+            "incrementAndGet\$atomicfu\$long" -> {
+                code = "(function(scope) {return $f = $f.inc();})"
+            }
+
+            "decrementAndGet\$atomicfu" -> {
+                code = "(function(scope) {return --$f;})"
+            }
+
+            "decrementAndGet\$atomicfu\$long" -> {
+                code = "(function(scope) {return $f = $f.dec();})"
+            }
+        }
+        if (code != null) {
+            this.getNode(code)
+            return true
+        }
+        return false
+    }
+
+    private fun FunctionCall.getNode(code: String) {
+        val p = Parser(CompilerEnvirons())
+        val node = p.parse(code, null, 0)
+        if (node.firstChild != null) {
+            val expr = (node.firstChild as ExpressionStatement).expression
+            val func = (expr as ParenthesizedExpression).expression as FunctionNode
+            (node.firstChild as ExpressionStatement).expression = ParenthesizedExpressionDerived(FunctionNodeDerived(func))
+            this.target = (node.firstChild as ExpressionStatement).expression
+            val thisNode = Parser(CompilerEnvirons()).parse("this", null, 0)
+            this.arguments = listOf((thisNode.firstChild as ExpressionStatement).expression)
+        }
+    }
+}
+
+private class ParenthesizedExpressionDerived(val expr: FunctionNode) : ParenthesizedExpression() {
+    override fun toSource(depth: Int): String = "(" + expr.toSource(0) + ")"
+}
+
+// local FunctionNode parser for atomic operations to avoid internal formatting
+private class FunctionNodeDerived(val fn: FunctionNode) : FunctionNode() {
+
+    override fun toSource(depth: Int): String {
+        val sb = StringBuilder()
+        sb.append("function")
+        sb.append("(")
+        printList(fn.params, sb)
+        sb.append(") ")
+
+        sb.append("{")
+        (fn.body as Block).forEach {
+            val sbStmt = StringBuilder()
```

Comment:
To much repetition of `sbStmt.append` in the below code. Please use `buildString { ... }` from the Kotlin stdlib. And you don't need `sbStmt` at all. All you do when you build a statement is that you append to top-level `sb`, so please directly use `buildString` at the very top level of this function like this:

```
override fun toSource(depth: Int) = buildString { 
    // ... code here ... 
}
```